CMakeLists.txt

```diff
@@ -35,6 +35,11 @@ else()
     endif()
 endif()
 
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic") # if compiler is GCC or Clang add warning flags
+endif()
+
```

Comment:
I think setting warning flags is a faux pas in a library project like CPR. Most people use this library inside of another cmake project where they are setting their own warning flags.

Perhaps a good compromise is to set these warning flags in the travis build config so that we can see the warnings in those builds? Let me know what you think.

In either case I'm unlikely to merge in this PR if it's setting warning flags like this.